arviz/data/inference_data.py

```diff
@@ -173,6 +175,56 @@ def __delattr__(self, group):
     def _groups_all(self):
         return self._groups + self._groups_warmup
 
+    def _extend_xr_method(func):  # pylint: disable=no-self-argument
```

Comment:
I think self should also go here

Reply:
I don't think this will work like this. I think as `_extend_xr_method` is a helper function, this can be defined outside the class(maybe in `base.py`)?

Reply:
I don't know, maybe ask in slack to get some advise on this? I am confused by the fact that the `wrapper` uses `_group_names` which is a private method (and I think it should be) so it should be part of idata class, maybe the wrapped method definitions should go in `__init__`? :confused: